src/components/DistanceRequest/index.js

```diff
@@ -0,0 +1,235 @@
+                    scrollEventThrottle={variables.distanceScrollEventThrottle}
+                    ref={scrollViewRef}
+                    renderItem={({item, drag, isActive, getIndex}) => (
+                        <DistanceRequestRenderItem
+                            waypoints={waypoints}
+                            item={item}
+                            onSecondaryInteraction={drag}
+                            isActive={isActive}
+                            getIndex={getIndex}
+                            onPress={navigateToWaypointEditPage}
+                            disabled={isLoadingRoute}
+                        />
+                    )}
+                    ListFooterComponent={
+                        <DistanceRequestFooter
+                            waypoints={waypoints}
+                            hasRouteError={hasRouteError}
+                            navigateToWaypointEditPage={navigateToWaypointEditPage}
+                            transactionID={transactionID}
+                        />
+                    }
+                />
+            </View>
+            <View style={[styles.w100, styles.pt2]}>
+                <Button
+                    success
+                    allowBubble
+                    pressOnEnter
+                    style={[styles.w100, styles.mb4, styles.ph4, styles.flexShrink0]}
+                    onPress={() => onSubmit(waypoints)}
+                    isDisabled={_.size(validatedWaypoints) < 2 || (!isOffline && (hasRouteError || isLoadingRoute || isLoading))}
+                    text={translate(isEditingRequest ? 'common.save' : 'common.next')}
+                    isLoading={!isOffline && (isLoadingRoute || shouldFetchRoute || isLoading)}
+                />
+            </View>
+        </>
+    );
+
+    if (!isEditing) {
+        return content;
+    }
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableKeyboardAvoidingView={false}
+            testID={DistanceRequest.displayName}
+        >
+            {({safeAreaPaddingBottomStyle}) => (
+                <FullPageNotFoundView shouldShow={!IOUUtils.isValidMoneyRequestType(iouType)}>
+                    <View style={[styles.flex1, safeAreaPaddingBottomStyle]}>
+                        <HeaderWithBackButton
+                            title={translate('common.distance')}
+                            onBackButtonPress={navigateBack}
+                        />
+                        {content}
+                    </View>
+                </FullPageNotFoundView>
+            )}
+        </ScreenWrapper>
+    );
+}
+
+DistanceRequest.displayName = 'DistanceRequest';
+DistanceRequest.propTypes = propTypes;
+DistanceRequest.defaultProps = defaultProps;
+export default withOnyx({
+    transaction: {
+        key: ({transactionID}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID || 0}`,
+    },
+    mapboxAccessToken: {
```

Comment:
I think this may have been missed during the refactoring, but this prop is no accessed in this component.

Reply:
yes, that can be removed as unused. Moved to DistanceRequestFooter

Reply:
That's right. I will prepare a PR for that tomorrow.

Reply:
Added this change to [this PR](https://github.com/Expensify/App/pull/28922/files).

cc @tgolen, @situchan, @blazejkustra